# ATIVIDADE PRÁTICA - CIÊNCIA DE DADOS

### Aluno: Pablo Henrique Cardoso Ramos | Data: 16/03/2026

Colocar em prática os conceitos de Ciência de Dados e Aprendizagem de
Máquina vistos em aula. Você deverá buscar um dataset real, realizar o pré-
processamento dos dados, aplicar um modelo de Regressão Linear e
interpretar os resultados usando as métricas R² e MSE.

### IMPORTAÇÃO DAS BIBLIOTECAS

In [ ]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score
import plotly.express as px

In [13]:
dados = pd.read_csv("train.csv")

In [3]:
display(dados)

,Id,MSSubClass,MSZoning,LotFrontage,LotArea,Street,Alley,LotShape,LandContour,Utilities,...,PoolArea,PoolQC,Fence,MiscFeature,MiscVal,MoSold,YrSold,SaleType,SaleCondition,SalePrice
0,1,60,RL,65.0,8450,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,NaN,NaN,0,2,2008,WD,Normal,208500
1,2,20,RL,80.0,9600,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,NaN,NaN,0,5,2007,WD,Normal,181500
2,3,60,RL,68.0,11250,Pave,NaN,IR1,Lvl,AllPub,...,0,NaN,NaN,NaN,0,9,2008,WD,Normal,223500
3,4,70,RL,60.0,9550,Pave,NaN,IR1,Lvl,AllPub,...,0,NaN,NaN,NaN,0,2,2006,WD,Abnorml,140000
4,5,60,RL,84.0,14260,Pave,NaN,IR1,Lvl,AllPub,...,0,NaN,NaN,NaN,0,12,2008,WD,Normal,250000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1455,1456,60,RL,62.0,7917,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,NaN,NaN,0,8,2007,WD,Normal,175000
1456,1457,20,RL,85.0,13175,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,MnPrv,NaN,0,2,2010,WD,Normal,210000
1457,1458,70,RL,66.0,9042,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,GdPrv,Shed,2500,5,2010,WD,Normal,266500
1458,1459,20,RL,68.0,9717,Pave,NaN,Reg,Lvl,AllPub,...,0,NaN,NaN,NaN,0,4,2010,WD,Normal,142125


In [4]:
dados.info()

<class 'pandas.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   str    
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   str    
 6   Alley          91 non-null     str    
 7   LotShape       1460 non-null   str    
 8   LandContour    1460 non-null   str    
 9   Utilities      1460 non-null   str    
 10  LotConfig      1460 non-null   str    
 11  LandSlope      1460 non-null   str    
 12  Neighborhood   1460 non-null   str    
 13  Condition1     1460 non-null   str    
 14  Condition2     1460 non-null   str    
 15  BldgType       1460 non-null   str    
 16  HouseStyle     1460 non-null   str    
 17  OverallQual    1460 non-null   int64  
 18  OverallCond    1460

In [5]:
colunas_analise = ['LotArea','OverallQual', 'YrSold', 'OverallCond', 'SalePrice']


df_filtrado = dados[colunas_analise]

display(df_filtrado)

,LotArea,OverallQual,YrSold,OverallCond,SalePrice
0,8450,7,2008,5,208500
1,9600,6,2007,8,181500
2,11250,7,2008,5,223500
3,9550,7,2006,5,140000
4,14260,8,2008,5,250000
...,...,...,...,...,...
1455,7917,6,2007,5,175000
1456,13175,6,2010,6,210000
1457,9042,7,2010,9,266500
1458,9717,5,2010,6,142125


In [6]:
df_filtrado.info()

<class 'pandas.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   LotArea      1460 non-null   int64
 1   OverallQual  1460 non-null   int64
 2   YrSold       1460 non-null   int64
 3   OverallCond  1460 non-null   int64
 4   SalePrice    1460 non-null   int64
dtypes: int64(5)
memory usage: 57.2 KB


In [7]:
display(df_filtrado.isnull().sum())

LotArea        0
OverallQual    0
YrSold         0
OverallCond    0
SalePrice      0
dtype: int64

# ETAPA 3: MINERAÇÃO DOS DADOS

In [8]:
X = df_filtrado[['LotArea', 'OverallQual', 'YrSold', 'OverallCond']]
y = df_filtrado['SalePrice'] 

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = LinearRegression()
model.fit(X_train, y_train)

y_pred = model.predict(X_test)


# ETAPA 4: INTERPRETAÇÃO E AVALIAÇÃO

In [9]:
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print(f"\nMSE (ERRO QUADRÁTICO MÉDIO): {mse:.2f}")
print(f"R² (COEFICIENTE DE DETERMINAÇÃO): {r2:.2f}")


MSE (ERRO QUADRÁTICO MÉDIO): 2491108418.27
R² (COEFICIENTE DE DETERMINAÇÃO): 0.68


# ETAPA 5: VISUALIZAÇÃO

In [12]:
fig = px.scatter(x=y_test, y=y_pred)
fig.update_layout(
    title="Previsão de Vendas com Regressão Linear",
    xaxis_title="Vendas Reais",
    yaxis_title="Vendas Previstas",
)
fig.show()

# PERGUNTAS SOBRE O MODELO CONSTRUÍDO:

1 - O modelo performou melhor ou pior do que o exemplo da loja de
departamentos do slide? Por quê?

Resposta: O modelo performou melhor, por conta do R², onde no do slide teve o valor de 0.01. No modelo construído, o R² foi de 0.68, ou seja, o modelo teve uma melhor performance, creio que a causa também possa ser o maior volume de dados.


2 - O R² foi próximo de 0 ou de 1?

Resposta: O R² foi mais próximo de 1.


3 - Analisando o gráfico: os pontos estão distribuídos próximos à linha
diagonal? O modelo consegue prever bem os preços mais altos e mais
baixos?

Resposta: Sim, os pontos estão bem distribuídos. Com os preços mais baixos creio que o modelo não performou muito bem, sendo em alguns casos como casas de 50k a 100k, o modelo acabou prevendo valores muito abaixo da realidade. Já com preços mais altos ele não conseguiu prever bem.